### ISSUE TITLE
ion-datetime returns different values when running on Android vs running on web and iOS

### ISSUE THREAD

**Issue by carstenbaumhoegger** [ionic-team/ionic#17720]
# Bug Report

**Ionic version:**

[x] **4.x**

**Current behavior:**

I'm using `ion-datetime` to let the user select a specific date in the future.

Android users were telling me about strange behaviour of the app if they were selecting the last possible date of a given date range (the user can only select day, month and year).

After some time of debugging I noticed this strange behaviour:

On iOS and web `ion-datetime` returns a date object with the time value at which component itself was created.
On Android `ion-datetime` returns the time value at which the actual change did happen.

e.g. the user navigates to page x at `2019-03-07T21:00:00.000-00:00`.

Now he waits exactly 2 minutes until he selects a date (2019-03-21) with an `ion-datetime`.

On iOS and web the returned value is:  `2019-03-21T21:00:00.000-00:00`

On Android the value is : `2019-03-21T21:02:00.000-00:00`

**Expected behavior:**

All platforms behave in a consitent way.

**Steps to reproduce:**

Use `ion-datetime` on different platforms and see, that the values returned are not consistent.

**Related code:**

here's my `ion-datetime` view-code
```html
            <ion-datetime [(ngModel)]="returnDate"
                          displayFormat="DD.MM.YYYY"
                          name="returnDate"
                          id="returnDate"
                          [yearValues]="yearValues"
                          (ionChange)="onDateChange($event)"
                          required></ion-datetime>
```

**Other information:**

**Ionic info:**

```
Ionic:

   ionic (Ionic CLI)             : 4.11.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.1.1
   @angular-devkit/build-angular : 0.13.5
   @angular-devkit/schematics    : 7.3.5
   @angular/cli                  : 7.3.1
   @ionic/angular-toolkit        : 1.3.0

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : android 7.1.4, ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 4.0.0, (and 11 other plugins)

System:

   ios-deploy : 1.9.4
   NodeJS     : v10.15.0 (/usr/local/bin/node)
   npm        : 6.8.0
   OS         : macOS Mojave
   Xcode      : Xcode 10.1 Build version 10B61
```

**Comment by bipoza** [comment]
I have suffered the same problem. Internal date ISO format on IOS is different than Android, I don't know why.
Finally i fixed this issue using moment.js, it is very simple. 

### Here's my date TypeScript
```
// Install module
npm install moment

// Import the module on the page you want to use
import * as moment from 'moment';

// Declare variable
public date;

// Initialize to the current date
this.date = moment().format();

// Parse edited date to ISO String
this.date = moment(this.date).toISOString();
```

### Here's my ion-datetime view-code

`<ion-datetime displayFormat="YYYY-MM-DD HH:mm" [(ngModel)]="date"></ion-datetime>`

With this we will get the same dates on both devices

**Comment by liamdebeasi** [comment]
Hi there,

Thanks for opening an issue with us! I merged in #17706 recently which fixes a few datetime issues. I created a nightly build for this and was wondering if you would be able to test your app with this to see if it fixes your issue? You can install the nightly using `npm i @ionic/core@dev`.

Otherwise, if you could provide a repository with the code required to reproduce this issue, that would be great!

Thanks for using Ionic!
